solver/cvrp.py

```diff
@@ -249,13 +248,15 @@ def solve_tsp_heuristic(self) -> None:
         for vehicle_id, cluster in clusters.items():
             idx = {id: i for i, id in enumerate(cluster)}
 
-            weight_matrix = np.zeros((len(cluster), len(cluster)))
+            G = nx.MultiDiGraph()
+            G.add_nodes_from(range(len(cluster)))
             for coord in combinations(cluster, 2):
                 coord_reverse = tuple(reversed(coord))
-                weight_matrix[idx[coord[0]], idx[coord[1]]] = self.costs[coord]
-                weight_matrix[idx[coord[1]], idx[coord[0]]] = self.costs[coord_reverse]
+                G.add_edge(idx[coord[0]], idx[coord[1]], weight=self.costs[coord])
+                G.add_edge(idx[coord[1]], idx[coord[0]], weight=self.costs[coord_reverse])
 
-            path, _ = solve_tsp_local_search(weight_matrix)
+            greedy_tsp = nx.approximation.traveling_salesman.greedy_tsp
+            path = nx.approximation.traveling_salesman_problem(G, cycle=False, method=greedy_tsp)
```

Comment:
Worth noting here (as in this PR, not in the code) is the use of a greedy algorithm and that it doesn't seem to affect the end-results _too_ much due to the way we use clustering for DQM and classical. The non-greedy methods took far too long to finish.

Reply:
added to the description for future reference 👍